lib/ImageResize.php

```diff
@@ -7,316 +7,333 @@
-
+    
+    /**
+     * @var bool
+     */
+    public $gamma_correction = false;
+    
+    /**
+     * @var int
+     */
     public $interlace = 1;
-
+    
+    /**
+     * @var mixed
+     */
     public $source_type;
-
+    
     protected $source_image;
-
+    
     protected $original_w;
     protected $original_h;
-
+    
     protected $dest_x = 0;
     protected $dest_y = 0;
-
+    
     protected $source_x;
     protected $source_y;
-
+    
     protected $dest_w;
     protected $dest_h;
-
+    
     protected $source_w;
     protected $source_h;
-
+    
     protected $source_info;
-
+    
     protected $filters = [];
-
-    /**
-     * Create instance from a strng
-     *
-     * @param string $image_data
-     * @return ImageResize
-     * @throws ImageResizeException
-     */
+    
     public static function createFromString($image_data)
     {
-        if (empty($image_data) || $image_data === null) {
-            throw new ImageResizeException('image_data must not be empty');
+        if (empty( $image_data ) || $image_data === null) {
+            throw new ImageResizeException( __CLASS__ . ' ERROR: image_data must not be empty.' );
         }
-        $resize = new self('data://application/octet-stream;base64,' . base64_encode($image_data));
-        return $resize;
+        return new self( 'data://application/octet-stream;base64,'.base64_encode( $image_data ) );
     }
-
-
-    /**
-     * Add filter function for use right before save image to file.
-     *
-     * @param callable $filter
-     * @return $this
-     */
-    public function addFilter(callable $filter)
-    {
-        $this->filters[] = $filter;
-        return $this;
-    }
-
-    /**
-     * Apply filters.
-     *
-     * @param $image resource an image resource identifier
-     * @param $filterType filter type and default value is IMG_FILTER_NEGATE
-     */
-    protected function applyFilter($image, $filterType = IMG_FILTER_NEGATE)
-    {
-        foreach ($this->filters as $function) {
-            $function($image, $filterType);
-        }
-    }
-
-    /**
-     * Loads image source and its properties to the instanciated object
-     *
-     * @param string $filename
-     * @return ImageResize
-     * @throws ImageResizeException
-     */
+    
     public function __construct($filename)
     {
-        if (!defined('IMAGETYPE_WEBP')) {
-            define('IMAGETYPE_WEBP', 18);
-        }
-        if ($filename === null || empty($filename) || (substr($filename, 0, 5) !== 'data:' && !is_file($filename))) {
-            throw new ImageResizeException('File does not exist');
-        }
-
-        $finfo = finfo_open(FILEINFO_MIME_TYPE);
-        $checkWebp = false;
-        if (strstr(finfo_file($finfo, $filename), 'image') === false) {
-            if (version_compare(PHP_VERSION, '7.0.0', '<=') && strstr(file_get_contents($filename), 'WEBPVP8') !== false) {
-                $checkWebp = true;
+        if (!defined( 'IMAGETYPE_WEBP' )) {
```

Comment:
Please consider reverting the previous coding style.

Reply:
code stylesheet required